### PR TITLE
Move the miopen_batch_norm decomposition to the core table

### PR DESCRIPTION
**Pull request by pablo-garay**
DTensor's sharding propagator has no strategy for `aten.miopen_batch_norm`, so `F.batch_norm` on a sharded 4D input fails on ROCm with "Operator aten.miopen_batch_norm.default does not have a sharding strategy registered". CUDA never hits the equivalent because `aten.cudnn_batch_norm` has a decomposition in `torch/_decomp`, which DTensor's DecompShardingStrategy falls back to. The ROCm decomposition existed only in `torch/_inductor/decomposition.py`, so inductor could use it but DTensor (and any other consumer of the core table) could not.

This moves the function unchanged into `torch/_decomp/decompositions.py` next to `cudnn_batch_norm`, registers it in the core decomposition list, and removes the inductor copy; inductor's table is built from the core table so it keeps the decomposition. The DTensor normalization test loses its ROCm skip.

Test plan, on gfx950 (ROCm 10.0), 4 ranks: both instantiations of the DTensor test pass (they raised NotImplementedError before). Inductor sanity: a Conv2d+BatchNorm2d+ReLU model compiled in train and eval mode matches eager and backward runs, and the decomposition is present in both the core and inductor tables.

    python -m pytest test/distributed/tensor/test_math_ops.py -k test_normalization_ops

gfx942/gfx90a validation via the ciflow labels below (the test is in the distributed config, so the periodic labels are the ones that run it); the inductor side is covered by the regular CUDA and ROCm inductor jobs.

This PR was authored with the assistance of an AI agent (Claude). The analysis and test results were verified by hand.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

**Review by jeffdaily — APPROVED** [review]

**Comment by jeffdaily** [comment]
These failures are not related to the PR content.

@pytorchbot merge -i
